### PR TITLE
Add NLP with Ruby

### PR DESCRIPTION
**Pull request by arbox**
**https://github.com/arbox/nlp-with-ruby**

**This list encompasses Ruby related projects for processing natural language texts. It it an attempt to create a curated list with only working software and no dead links. The list is focused only on Ruby so it does not duplicate available lists.**

# By submitting this pull request I confirm I've read and complied with the below requirements.

Failure to properly do so will just result in the pull request being closed and everyone's time wasted. Please read it twice. Most people miss many things.

- I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title. For example, `Add Name of List`, not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should include a short description about the list project/theme (not the list itself), and should be added at the bottom of the appropriate category.
- The list I added
	- **has been around for at least 30 days,**
	- is the result of hard work and is the best I could possibly produce,
	- is a non-generated Markdown file in a GitHub repo,
	- is not a duplicate,
	- includes a succinct description of the project/theme at the top ([example](https://github.com/willempienaar/awesome-quantified-self)),
	- includes a project logo/illustration (whenever possible) at the top right of the readme ([example](https://github.com/sindresorhus/awesome-electron)). (The logo should link to the project website or any relevant website. The logo should be high-DPI, so set it to maximum half the width of the original image.),
	- only has awesome items **with descriptions** *(Awesome lists are curations of the best, not of everything)*,
	- contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
	- has a Table of Contents section **named `Contents`** as the first section,
	- has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) *(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)* *(If you use a license badge, it should be SVG, not PNG)*,
	- has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
	- doesn't include a Travis badge,
	- has been checked for inconsistent formatting, spelling, and grammar:
		- Each link description starts with an uppercase character and ends with a period.
		- Drop all the `A` / `An` prefixes in the descriptions.
		- Consistent naming. For example, `Node.js`, not `NodeJS` or `node.js`.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Some feedback:

- Don't put newline between list items.
- Use `-` for lists, not `*`.
- `StackOverflow` => `Stack Overflow` and the link should probably be: http://stackoverflow.com/questions/tagged/nlp+ruby
- Why the `nlcproc` search on Twitter? Should be clearer for the user what to expect. I expected Ruby related NLP tweets.
- `Useful resources for text processing in Ruby` is misleading. NLP is not plainly text processing.
- `Please read the Contributors Guide and refer the Contribution section.` to point in linking to both. Why even have a `Contribution` section then that just links to the guide.
- The repo title is very long, I would have done it the opposite way. Made the repo title use `NLP` and the readme title use the full title. Would also look better if the repo title was lowercased like most other repos.
- The top description should talk a little bit more what NLP is and link to a resource for the user can read more about NLP.

File: readme.md [hunk 1]
```diff
@@ -226,6 +226,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
 	- [Tutorials](https://github.com/ujjwalkarn/Machine-Learning-Tutorials)
 - [Speech and Natural Language Processing](https://github.com/edobashira/speech-language-processing)
 	- [Spanish](https://github.com/dav009/awesome-spanish-nlp)
+	- [NLP with Ruby](https://github.com/arbox/Awesome-Natural-Language-Processing-with-Ruby) - Ruby focused NLP tools and resources.
```
Comment (sindresorhus) [inline readme.md]: `tools and resources` are moot. Almost all lists have these. I don't think it needs a description, as it's just a repetition of the name anyways.

**Comment by arbox** [comment]
Thank you for the valuable feedback, indeed there are many points we can improve!

> Don't put newline between list items.

Addressed.

> Use - for lists, not *.

Changed.

> StackOverflow => Stack Overflow and the link should probably be: http://stackoverflow.com/questions/tagged/nlp+ruby

I've expanded the search string to cover only Ruby and NLP specific questions.

> Why the nlproc search on Twitter? Should be clearer for the user what to expect. I expected Ruby related NLP tweets.

`#nlproc` is the standard hash tag for NLP related messages. But your objections are correct, I elaborated the query so we accept now a much stricter selection.
 
> `Useful resources for text processing in Ruby` is misleading. NLP is not plainly text processing.

Do you have any suggestions? I see your point that NLP is not text processing as in `MS Word is a text processor.` but processing is a very broad term, and what is processed are texts (spoken or written). Text is more or less clearly defined in Linguistics. 

> `Please read the Contributors Guide and refer the Contribution section.` to point in linking to both. Why even have a Contribution section then that just links to the guide.

I have to change this. But some duplication is useful since people do not read guidelines, being redundant doesn't hurt. 

> The repo title is very long. I would have done it the opposite way. Made the repo title use NLP and the readme title use the full title. Would also look better if the repo title was lowercased like most other repos. Preferred repo title: nlp-with-ruby

You're right, the way round is better! Changed.

> The top description should talk a little bit more what NLP is and link to a resource for the user can read more about NLP.

Actually  I don't see the point in general NLP introduction at this point. Users should be familiar with both NLP and Ruby before they look for Ruby focused NLP resources. But I'll include a short refinement to distinguish this list from `Neuro-linguistic Programming`  which is very notorious on the Web.

Thank you again and please review my changes :)

**Comment by arbox** [comment]
@sindresorhus could you please review the changes?

**Comment by arbox** [comment]
@sindresorhus Thank you!

**Pull request merged**